Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/calculation/CclJsonFunctions.kt

```diff
@@ -0,0 +1,53 @@
+package de.rki.coronawarnapp.ccl.dccwalletinfo.calculation
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import de.rki.coronawarnapp.ccl.configuration.model.CCLConfiguration
+import de.rki.coronawarnapp.ccl.configuration.storage.CCLConfigurationRepository
+import de.rki.coronawarnapp.util.coroutine.AppScope
+import de.rki.coronawarnapp.util.serialization.BaseJackson
+import de.rki.jfn.JsonFunctions
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class CclJsonFunctions @Inject constructor(
+    @BaseJackson private val mapper: ObjectMapper,
+    @AppScope private val appScope: CoroutineScope,
+    private val configurationRepository: CCLConfigurationRepository,
+) {
+
+    private lateinit var jsonFunctions: JsonFunctions
+
+    init {
+        appScope.launch {
```

Comment:
Why is the `appScope` necessary here?

Reply:
to get the configuration you need a coroutine

Reply:
would you prefer using `runBlocking` here ?

Reply:
Dagger is creating its object graph on the Main Thread, right - so with the `appScope` the performance would be a bit better since the Main Thread wouldn't need to wait for data from the repo.

Reply:
So, I leave as is